src/utils/components/DiskModal/components/StorageProfileSettings/VolumeMode.tsx

```diff
@@ -1,58 +1,69 @@
-import React, { FC, useEffect, useMemo } from 'react';
-import { Controller, useFormContext } from 'react-hook-form';
+import React, { FC, useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { uniq } from 'lodash';
 
-import {
-  V1beta1StorageSpecAccessModesEnum,
-  V1beta1StorageSpecVolumeModeEnum,
-} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { V1beta1StorageSpecVolumeModeEnum } from '@kubevirt-ui/kubevirt-api/kubevirt';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { ClaimPropertySets } from '@kubevirt-utils/types/storage';
 import { FormGroup, Radio } from '@patternfly/react-core';
 
 import { V1DiskFormState } from '../../utils/types';
-import {
-  ACCESS_MODE_FIELD,
-  STORAGE_CLASS_PROVIDER_FIELD,
-  VOLUME_MODE_FIELD,
-  VOLUMEMODE_FIELDID,
-} from '../utils/constants';
-import { getVolumeModeForProvisioner, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
+import { ACCESS_MODE_FIELD, VOLUME_MODE_FIELD, VOLUMEMODE_FIELDID } from '../utils/constants';
+import { getAccessModesForVolume, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
 
-const VolumeMode: FC = () => {
+import RecommendationLabel from './RecommendationLabel';
+
+type VolumeModeProps = {
+  allowAllModes: boolean;
+  claimPropertySets: ClaimPropertySets;
+};
+
+const VolumeMode: FC<VolumeModeProps> = ({ allowAllModes, claimPropertySets }) => {
   const { t } = useKubevirtTranslation();
 
-  const { control, setValue, watch } = useFormContext<V1DiskFormState>();
+  const { setValue, watch } = useFormContext<V1DiskFormState>();
 
   const volumeMode = watch(VOLUME_MODE_FIELD);
-  const accessModes = watch(ACCESS_MODE_FIELD) as V1beta1StorageSpecAccessModesEnum[];
-  const storageClassProvisioner = watch(STORAGE_CLASS_PROVIDER_FIELD);
-
-  const allowedVolumeModes = useMemo(
-    () => getVolumeModeForProvisioner(storageClassProvisioner, accessModes?.[0]),
-    [accessModes, storageClassProvisioner],
+  const recommendedVolumeModes = uniq(
+    claimPropertySets
+      .map((it) => it.volumeMode)
+      .filter(Boolean)
+      .map((mode): V1beta1StorageSpecVolumeModeEnum => V1beta1StorageSpecVolumeModeEnum[mode])
+      .filter(Boolean),
```

Comment:
add optional chaining in case claimPropertySets is undefined

Reply:
At the component level it should be well defined (may be empty). The typing reflects that. The fallback it's handled at the parent level i.e.
```
<AccessMode allowAllModes={allowAllModes} claimPropertySets={claimPropertySets ?? []} />
```

Reply:
OK. I've just realized we are compiling with `strict: false`.